Mathlib/Order/Antisymmetrization.lean

```diff
@@ -112,9 +128,86 @@ end IsPreorder
 
-variable [Preorder α] [Preorder β] {a b : α}
+variable [Preorder α] [Preorder β]
+
+theorem le_iff_lt_or_antisymmRel : a ≤ b ↔ a < b ∨ AntisymmRel (· ≤ ·) a b := by
+  rw [lt_iff_le_not_le, antisymmRel_iff]
+  tauto
+
+alias ⟨LE.le.lt_or_antisymmRel, _⟩ := le_iff_lt_or_antisymmRel
+
+@[trans]
+theorem le_of_le_of_antisymmRel (h₁ : a ≤ b) (h₂ : AntisymmRel (· ≤ ·) b c) : a ≤ c :=
+  h₁.trans h₂.le
+
+alias LE.le.trans_antisymmRel := le_of_le_of_antisymmRel
+
+instance : @Trans α α α (· ≤ ·) (AntisymmRel (· ≤ ·)) (· ≤ ·) where
+  trans := le_of_le_of_antisymmRel
+
+@[trans]
+theorem le_of_antisymmRel_of_le (h₁ : AntisymmRel (· ≤ ·) a b) (h₂ : b ≤ c) : a ≤ c :=
+  h₁.le.trans h₂
+
+alias AntisymmRel.trans_le := le_of_antisymmRel_of_le
+
+instance : @Trans α α α (AntisymmRel (· ≤ ·)) (· ≤ ·) (· ≤ ·) where
+  trans := le_of_antisymmRel_of_le
+
+@[trans]
+theorem lt_of_lt_of_antisymmRel (h₁ : a < b) (h₂ : AntisymmRel (· ≤ ·) b c) : a < c :=
+  h₁.trans_le h₂.le
+
+alias LT.lt.trans_antisymmRel := lt_of_lt_of_antisymmRel
+
+instance : @Trans α α α (· < ·) (AntisymmRel (· ≤ ·)) (· < ·) where
+  trans := lt_of_lt_of_antisymmRel
+
+@[trans]
+theorem lt_of_antisymmRel_of_lt (h₁ : AntisymmRel (· ≤ ·) a b) (h₂ : b < c) : a < c :=
+  h₁.le.trans_lt h₂
+
+alias AntisymmRel.trans_lt := lt_of_antisymmRel_of_lt
+
+instance : @Trans α α α (AntisymmRel (· ≤ ·)) (· < ·) (· < ·) where
+  trans := lt_of_antisymmRel_of_lt
+
+theorem AntisymmRel.le_congr (h₁ : AntisymmRel (· ≤ ·) a b) (h₂ : AntisymmRel (· ≤ ·) c d) :
+    a ≤ c ↔ b ≤ d where
+  mp h := (h₁.symm.trans_le h).trans_antisymmRel h₂
+  mpr h := (h₁.trans_le h).trans_antisymmRel h₂.symm
+
+theorem AntisymmRel.le_congr_left (h : AntisymmRel (· ≤ ·) a b) : a ≤ c ↔ b ≤ c :=
+  h.le_congr (antisymmRel_refl _ c)
+
+theorem AntisymmRel.le_congr_right (h : AntisymmRel (· ≤ ·) b c) : a ≤ b ↔ a ≤ c :=
+  (antisymmRel_refl _ a).le_congr h
+
+theorem AntisymmRel.lt_congr (h₁ : AntisymmRel (· ≤ ·) a b) (h₂ : AntisymmRel (· ≤ ·) c d) :
+    a < c ↔ b < d where
+  mp h := (h₁.symm.trans_lt h).trans_antisymmRel h₂
+  mpr h := (h₁.trans_lt h).trans_antisymmRel h₂.symm
+
+theorem AntisymmRel.lt_congr_left (h : AntisymmRel (· ≤ ·) a b) : a < c ↔ b < c :=
+  h.lt_congr (antisymmRel_refl _ c)
+
+theorem AntisymmRel.lt_congr_right (h : AntisymmRel (· ≤ ·) b c) : a < b ↔ a < c :=
+  (antisymmRel_refl _ a).lt_congr h
+
+theorem AntisymmRel.antisymmRel_congr
+    (h₁ : AntisymmRel (· ≤ ·) a b) (h₂ : AntisymmRel (· ≤ ·) c d) :
+      AntisymmRel (· ≤ ·) a c ↔ AntisymmRel (· ≤ ·) b d := by
+  rw [antisymmRel_iff, antisymmRel_iff, h₁.le_congr h₂, h₂.le_congr h₁]
+
+theorem AntisymmRel.antisymmRel_congr_left (h : AntisymmRel (· ≤ ·) a b) :
+    AntisymmRel (· ≤ ·) a c ↔ AntisymmRel (· ≤ ·) b c :=
+  h.antisymmRel_congr (antisymmRel_refl _ c)
+
+theorem AntisymmRel.antisymmRel_congr_right (h : AntisymmRel (· ≤ ·) b c) :
+    AntisymmRel (· ≤ ·) a b ↔ AntisymmRel (· ≤ ·) a c :=
+  (antisymmRel_refl _ a).antisymmRel_congr h
```

Comment:
Do we really want these lemmas? They are all consequences of symmetry and transitivity of `AntisymmRel (· ≤ ·)`.

Reply:
All of them are encoding an important fact: all of the order relations are preserved up to order equivalence.

Reply:
IMO they should be stated with `AntisymmRel (· ≤ ·)` replaced by `r` satisfying `IsSymm _ r` and `IsTrans _ r` and moved to that file.

Reply:
This is *just* for the `antisymmRel_congr` lemmas, right? Otherwise I think they'd be false: `a ≤ b` does not imply `a ≤ c ↔ b ≤ c`.

Do you have any file in mind?

Reply:
Yeah, the three lemmas are what I commented on. Is it not showing as such?
![image](https://github.com/user-attachments/assets/d4bf00da-bfc3-4e81-b2dd-dd169c7a4b48)

The file where IsSymm and IsTrans are defined is Mathlib.Order.Defs.Unbundled.